File: w1/w1d2/bubble_sort.rb
def bubble_sort(array)
  
  sorted = false
  until sorted
    sorted = true
    (array.count - 1).times do |i|
      if array[i] > array[i + 1]
        array[i], array[i + 1] = array[i + 1], array[i]
        sorted = false
      end  
    end
  end
  array  
end

File: w5/d4/bubble_sort.js
Array.prototype.bubbleSort = function () {
  var sorted = false;
  
  while(!sorted){
    sorted = true;
    
    for(var i = 0, j = 1; i < this.length - 1; ++i, ++j){
      if(this[i] > this[j]){
        var temp = this[i];
        this[i] = this[j];
        this[j] = temp;
        sorted = false;
      }
    }
  }
  return this;
};

var array = [1, 4, 7, 2, 8, 7, 3];

console.log(array.bubbleSort());